engine/engine_test.go

```diff
@@ -0,0 +1,76 @@
+package engine
+
+import (
+	"testing"
+
+	"github.com/thrasher-corp/gocryptotrader/config"
+)
+
+func Test_loadConfigWithSettings(t *testing.T) {
+	empty := ""
+	somePath := "somePath"
+	tests := []struct {
+		name     string
+		flags    []string
+		settings *Settings
+		want     *string
+		wantErr  bool
+	}{
+		{
+			name:     "empty",
+			settings: &Settings{},
+			wantErr:  true,
+		},
+		{
+			name: "invalid file",
+			settings: &Settings{
+				ConfigFile: "nonExistent.json",
+			},
+			wantErr: true,
+		},
+		{
+			name: "test file",
```

Comment:
I am also getting it here, which I don't understand, because the test output says it loads the testconfig, but the next line will attempt to load my real config's datadir:
```
== RUN   Test_loadConfigWithSettings/test_file
2020/09/11 10:07:58 Loading config file ../testdata/configtest.json..
Directory asdf\database does not exist.. creating.
...
    engine_test.go:71: loadConfigWithSettings() = asdf, want 
    --- FAIL: Test_loadConfigWithSettings/test_file (0.04s)
```

> configtest.json:
```
{
 "name": "Skynet",
 "encryptConfig": -1,
 "globalHTTPTimeout": 15000000000,
 },
```

>Realconfig:
```
{
 "name": "Skynet",
 "dataDirectory": "asdf",
 "encryptConfig": -1,
 "globalHTTPTimeout": 15000000000,
 },
```

To fix this, you'll need to update `testconfig.json` to include the `dataDirectory` property like you have to `config_example.json`

Reply:
Hmm... I can't 100% put a finger on it as I'm unable to reproduce, but I suspect it's to do with the shared package variables `config.Cfg`, `engine.flagSet` and (parallel?) tests. Then, same as above, this is interacting with your real config because the default config (loaded in `config.GetFilePath`) from some of the other tests leaks through to `config.Cfg` and wrong configuration is used as a result instead.
What is your exact command you are executing tests with? `make test`?

Some more thoughts/issues:
1. The current tests do already (before this change) create the default data dir paths if they don't exist (and don't clean them up). I'm not convinced setting the `dataDirectory` to a relative directory in the test config helps. Anyway, that'll create a lot of directories (in go packages) that don't get cleaned up after tests (versus just the one in the home dir at the moment).
2. why are `flagSet` and `config.Cfg` package level variables instead of proper function parameters? I've changed `engine.NewFromSettings()` to create a new config struct instead. Does that fix your test failure?
3. since some tests always fail on my local environment, it's difficult to see regressions unless I push an update, so forgive my small (not always completely tested) commits